planning/behavior_path_planner/src/scene_module/lane_change/external_request_lane_change_module.cpp

```diff
@@ -0,0 +1,752 @@
+    {selected_path.shift_line.start, selected_path.shift_line.end},
+    {output.start_distance_to_path_change, output.finish_distance_to_path_change},
+    SteeringFactor::LANE_CHANGE, steering_factor_direction, SteeringFactor::APPROACHING, "");
+}
+Pose ExternalRequestLaneChangeModule::getEgoPose() const { return planner_data_->self_pose->pose; }
+Twist ExternalRequestLaneChangeModule::getEgoTwist() const
+{
+  return planner_data_->self_odometry->twist.twist;
+}
+std_msgs::msg::Header ExternalRequestLaneChangeModule::getRouteHeader() const
+{
+  return planner_data_->route_handler->getRouteHeader();
+}
+void ExternalRequestLaneChangeModule::generateExtendedDrivableArea(PathWithLaneId & path)
+{
+  const auto & common_parameters = planner_data_->parameters;
+  const auto & route_handler = planner_data_->route_handler;
+  const auto drivable_lanes = lane_change_utils::generateDrivableLanes(
+    *route_handler, status_.current_lanes, status_.lane_change_lanes);
+  const auto expanded_lanes = util::expandLanelets(
+    drivable_lanes, parameters_->drivable_area_left_bound_offset,
+    parameters_->drivable_area_right_bound_offset);
+
+  util::generateDrivableArea(path, expanded_lanes, common_parameters.vehicle_length, planner_data_);
+}
+
+bool ExternalRequestLaneChangeModule::isApprovedPathSafe(Pose & ego_pose_before_collision) const
+{
+  const auto current_pose = getEgoPose();
+  const auto current_twist = getEgoTwist();
+  const auto & dynamic_objects = planner_data_->dynamic_object;
+  const auto & current_lanes = status_.current_lanes;
+  const auto & common_parameters = planner_data_->parameters;
+  const auto & route_handler = planner_data_->route_handler;
+  const auto path = status_.lane_change_path;
+
+  constexpr double check_distance = 100.0;
+  // get lanes used for detection
+  const double check_distance_with_path =
+    check_distance + path.preparation_length + path.lane_change_length;
+  const auto check_lanes = route_handler->getCheckTargetLanesFromPath(
+    path.path, status_.lane_change_lanes, check_distance_with_path);
+
+  std::unordered_map<std::string, CollisionCheckDebug> debug_data;
+
+  const size_t current_seg_idx = motion_utils::findFirstNearestSegmentIndexWithSoftConstraints(
+    path.path.points, current_pose, common_parameters.ego_nearest_dist_threshold,
+    common_parameters.ego_nearest_yaw_threshold);
+  return lane_change_utils::isLaneChangePathSafe(
+    path.path, current_lanes, check_lanes, dynamic_objects, current_pose, current_seg_idx,
+    current_twist, common_parameters, *parameters_,
+    common_parameters.expected_front_deceleration_for_abort,
+    common_parameters.expected_rear_deceleration_for_abort, ego_pose_before_collision, debug_data,
+    false, status_.lane_change_path.acceleration);
+}
+
+void ExternalRequestLaneChangeModule::updateOutputTurnSignal(BehaviorModuleOutput & output)
+{
+  const auto turn_signal_info = util::getPathTurnSignal(
```

Comment:
Since turn signal is no longer computed in candidate path, it might not appear here too.
Might need to add this [function](https://github.com/autowarefoundation/autoware.universe/blob/main/planning/behavior_path_planner/src/scene_module/lane_change/lane_change_module.cpp#:~:text=calcTurnSignalInfo()%3B)

Reply:
Done!
https://github.com/autowarefoundation/autoware.universe/pull/2442/commits/2fdc5feae3c974fcbeda9f61e48c02f83b1a9e16